Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -1877,25 +1882,25 @@ def __init__(self, paramSearch, paramReplace, verbose=False, whitelist=()):
             self._verbose = verbose
             self._whitelist = whitelist
 
-        def doIt(self,pset,base):
+        def doIt(self, pset, base):
             if isinstance(pset, cms._Parameterizable):
                 for name in pset.parameters_().keys():
                     # skip whitelisted parameters
                     if name in self._whitelist:
                         continue
                     # if I use pset.parameters_().items() I get copies of the parameter values
                     # so I can't modify the nested pset
-                    value = getattr(pset,name)
-                    type = value.pythonTypeName()
-                    if type in ('cms.PSet', 'cms.untracked.PSet'):
+                    value = getattr(pset, name)
+                    valueType = type(value)
+                    if valueType in [cms.EDAlias, cms.EDAnalyzer, cms.EDFilter, cms.EDProducer, cms.PSet, cms.untracked.PSet]:
```

Comment:
This change was needed because the routine didn't know how to handle a Sequence containing a module with a parameter like:
```py
[..]
    cpu = cms.EDProducer([..]
```
The syntax error was
```
    type = value.pythonTypeName()
AttributeError: 'EDProducer' object has no attribute 'pythonTypeName'
```
I'm not sure if other types should be taken into account, beyond the 4 I added (i.e. `cms.ED{Alias,Analyzer,Filter,Producer}`).

Reply:
> I'm not sure if other types should be taken into account, beyond the 4 I added (i.e. `cms.ED{Alias,Analyzer,Filter,Producer}`).

I'd say it would be sufficient to add only `EDProducer` of those, and the others can create confusion in the future.

The `EDProducer` and `EDAlias` can be brought in only with `SwitchProducer`. Currently there is nothing that could make `EDAnalyzer` or `EDFilter` available here, and at this stage I see it unlikely that would change.

This visitor seems to be only about changing the process name in `InputTag` and in `string`. `EDAlias` can only refer to products produced in the current process, and therefore changing process names there should have zero impact (I could also claim any change in EDAlias done by this visitor would be an error).

Reply:
Thanks for the insight. I will then make the following change, once the tests finish:
```diff
-if valueType in [cms.EDAlias, cms.EDAnalyzer, cms.EDFilter, cms.EDProducer, cms.PSet, cms.untracked.PSet]:
+if valueType in [cms.EDProducer, cms.PSet, cms.untracked.PSet]:
```